activesupport/lib/active_support/deprecation/reporting.rb

```diff
@@ -5,8 +5,6 @@ class Deprecation
     module Reporting
       # Whether to print a message (silent mode)
       attr_accessor :silenced
-      # Name of gem where method is deprecated
-      attr_accessor :gem_name
```

Comment:
Hardly it will break any behavior unless someone is including `Reporting` in their own class, what I'd not recommend anyway since this method depends on the others in Deprecation. I think this change is fine since the public API (`ActiveSupport::Deprecation`) did't changed.